internal/registration/internal/domain/user_password.go

```diff
@@ -0,0 +1,35 @@
+package domain
+
+import (
+	"errors"
+	"strings"
+)
+
+var ErrPasswordCannotBeEmpty = errors.New("password is empty")
+
+type UserPassword struct {
+	value string
+}
+
+func CreateUserPassword(value string) (UserPassword, error) {
+	if len(strings.TrimSpace(value)) == 0 {
+		return UserPassword{}, ErrPasswordCannotBeEmpty
+	}
+	return UserPassword{value}, nil
+}
+
+func (u UserPassword) String() string {
+	return u.value
+}
+
+type HashedUserPassword struct {
+	value string
+}
+
+func CreateHashedUserPassword(password UserPassword) HashedUserPassword {
```

Comment:
This function should return an (hashedPass, error) and accept the string. This conversion would be made by PasswordHasher implementation.